### PR TITLE
Update machine-controller to v1.11.1

### LINKED ISSUE #805: [vSphere] Update machine-controller version to include datastore cluster support
This issue is to bump machine-controller version used in kubeone to at least [v1.10.2](https://github.com/kubermatic/machine-controller/tree/v1.10.2) in order to have datastore cluster support for vSphere provider.

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

* Update `machine-controller` to v1.11.1
* Enable NodeCSRApprover controller
* Use `machine-controller` Cluster-API types and drop the Cluster-API dependency
* Update `machine-controller` types
* Update Terraform examples

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #805

**Does this PR introduce a user-facing change?**:
```release-note
Update `machine-controller` to v1.11.1
Enable NodeCSRApprover controller
Use `machine-controller` Cluster-API types and drop the Cluster-API dependency
Update `machine-controller` types
Update Terraform examples
```

/assign @kron4eg

### CODE REVIEW

File: examples/terraform/aws/output.tf [hunk 1]
```diff
@@ -72,7 +72,9 @@ output "kubeone_workers" {
           diskSize         = 50
           diskType         = "gp2"
           ## Only applicable if diskType = io1
-          diskIops = 500
+          diskIops         = 500
+          isSpotInstance   = false
+          ebsVolumeEncrypted = false
```
Comment (xmudrii) [inline examples/terraform/aws/output.tf]: @kron4eg Those are newly added API fields (`isSpotInstance` is older, but we added it just now). Should both be false?

Comment (kron4eg) [reply examples/terraform/aws/output.tf]: yes, and also what will be required to use `ebsVolumeEncrypted=true`? I guess some AWS KMS magic whatever, but still, question is open.

Comment (xmudrii) [reply examples/terraform/aws/output.tf]: I'm not really sure. I have assumed the same (regarding AWS KMS magic) and that's the reason I set it to false.

**Comment by kron4eg** [comment]
/hold

we need to figure out something about those E2E

**Pull request merged**